internal/kafka/internal/handlers/admin_kafka.go

```diff
@@ -169,13 +171,33 @@ func (h *adminKafkaHandler) Update(w http.ResponseWriter, r *http.Request) {
 				return false
 			}
 
+			getStatusBasedOnSuspendedParam := func(susp *bool, kafka *dbapi.KafkaRequest) string {
+				if shared.IsNilPredicate(susp) {
+					return kafka.Status
+				} else {
+					if *susp {
```

Comment:
This means we can update the status from `failed` to `suspending`. Do we want to do that?

Reply:
deleting isn't one of the [updatable statuses](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1344/files#diff-08f986360fba8bdf2eb71752a89a88a34d0c4a79014ec11c306ebd6586bed919R101). thinking now - should `deprovision` also be removed from the updatable statuses? this way we wouldn't need to add extra check here

Reply:
> should deprovision also be removed from the updatable statuses? this way we wouldn't need to add extra check here

We need to be able to update Strimzi version of a Kafka in deprovision status. See https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/787 for the motivation

Reply:
@machi1990 should we set to resuming if kafka is  in `KafkaRequestStatusDeprovision` or `KafkaRequestStatusDeleting` state?

Reply:
nvm. it will not happen. ignore

Reply:
@JameelB I have added your requested change. will squash the commits after CI passes

Reply:
Thanks @pawelpaszki! Seen the changes, look good to me :+1: 

There's an open question in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1346#discussion_r995497253 which may change the logic here.

Reply:
yes. saw that. will await further comments there